Automattic/wp-calypso
# Onboarding: Persist site intent

#### Changes proposed in this Pull Request

* Implement `setIntentOnSite` and attach to `apiRequestFunction` of `intent` step so that we can save the site intent after the user completes the hero flow

#### Testing instructions



* Apply D68408-code to your sandbox
* Ensure your API points to your sandbox
* Go to `/start/setup-site?siteSlug=<your_site>&flags=signup/hero-flow`
* Complete the hero flow either choosing `write` or `build` intent
* Check your network calling `/wpcom/v2/sites/<your_site>/site-intent` with the intent you selected



Related to https://github.com/Automattic/wp-calypso/issues/56568
